src/app/components/chat/jh-chat.directive.js

```diff
@@ -25,18 +27,95 @@
     };
 
     function jhChatLink(scope) {
-      scope.$watch('vm.messages.length', function(newVal) {
-        /* Update messages */
-        if (newVal !== undefined) {
-          // Scroll to bottom of messages list.
-          var messagesList = document.getElementById('jh-chat-messages');
-          setTimeout(function() {
-            messagesList.scrollTop = messagesList.scrollHeight;
-          }, 100);
+      scope.entriesCount = 0;       // number of all types of entries
+      scope.messagesCount = 0;      // number of entries of type 'chatMsg'
+      scope.lastSeenMessage = 0;    // last seen entry of type 'chatMsg'
+      scope.isChatVisible = false;
+
+      scope.$watch('vm.logEntries().length', function() {
+        // Scroll to bottom of messages list.
+        var messagesList = document.getElementById('chat-messages-box');
+        setTimeout(function() {
+          messagesList.scrollTop = messagesList.scrollHeight;
+        }, 100);
+      });
+
+      scope.$watch('vm.logEntries()', function(entries) {
+        // True only at the beginning when there are no chat messages
+        if (entries.length === 0) {
+          return;
+        }
+
+        // Update the chat messages count
+        for (var i = scope.entriesCount; i < entries.length; i++) {
+          if (entries[i].type === 'chatMsg') {
+            scope.messagesCount++;
+            if (scope.isChatVisible) {
+              scope.lastSeenMessage++;
+            } else {
+              var chatHeader = document.getElementById('chat-header');
+              chatHeader.innerHTML = scope.messagesCount - scope.lastSeenMessage;
+            }
+          } else {
```

Comment:
Why not just remove this else? You can put a similar comment on the `if` if you do not want to miss this information.